app/proc.go

```diff
@@ -3,18 +3,37 @@ package app
 import (
 	"context"
 	"raccoon/logger"
+	"raccoon/metrics"
+	"runtime"
 )
 
 //Run the server
 func Run() error {
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 
-	//@TODO - init config
+	registerCallback()
 
 	//start server
 	StartServer(ctx, cancel)
 	logger.Info("App.Run --> Complete")
 	<-ctx.Done()
 	return nil
 }
+
+func registerCallback() {
+	m := &runtime.MemStats{}
+	metrics.RegisterCallback(func() {
```

Comment:
explicitly calling of this as `callback` is confusing since the registered callback is not really an object that this client needs reference to in future. Would it make sense to call it likes of `registerMetricReporter` ? wdyt?